datarepo-clienttests/src/main/java/runner/TestRunner.java

```diff
@@ -0,0 +1,326 @@
+            // an exception
+            result = userJourneyFuture.get();
+            result.completed = true;
+          } catch (ExecutionException execEx) {
+            // user journey thread threw an exception and didn't populate its own return object
+            result = new UserJourneyResult(testScriptSpecification.name, "");
+            result.completed = false;
+            result.exceptionThrown = execEx;
+          }
+        else {
+          // user journey either was never started or got cancelled before it finished
+          result = new UserJourneyResult(testScriptSpecification.name, "");
+          result.completed = false;
+        }
+        results.add(result);
+      }
+    }
+    return results;
+  }
+
+  /**
+   * Call the setup() method of each TestScript class. If one of the classes throws an exception,
+   * stop looping through the remaining setup methods and return the exception.
+   *
+   * @return the exception thrown, null if none
+   */
+  Exception callTestScriptSetups() {
+    for (TestScript testScript : scripts) {
+      try {
+        testScript.setup(apiClientsForUsers);
+      } catch (Exception setupEx) {
+        // return the first exception thrown and stop looping through the setup methods
+        return setupEx;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Call the cleanup() method of each TestScript class. If any of the classes throws an exception,
+   * keep looping through the remaining cleanup methods before returning. Save the first exception
+   * thrown and return it.
+   *
+   * @return the first exception thrown, null if none
+   */
+  Exception callTestScriptCleanups() {
+    Exception exceptionThrown = null;
+    for (TestScript testScript : scripts) {
+      try {
+        testScript.cleanup(apiClientsForUsers);
+      } catch (Exception cleanupEx) {
+        // save the first exception thrown, keep looping through the remaining cleanup methods
+        // before returning
+        if (exceptionThrown == null) {
+          exceptionThrown = cleanupEx;
+        }
+      }
+    }
+    return exceptionThrown;
+  }
+
+  static class UserJourneyThread implements Callable<UserJourneyResult> {
+    TestScript testScript;
+    String testScriptName;
+    ApiClient apiClient;
+
+    public UserJourneyThread(TestScript testScript, String testScriptName, ApiClient apiClient) {
+      this.testScript = testScript;
+      this.testScriptName = testScriptName;
+      this.apiClient = apiClient;
+    }
+
+    public UserJourneyResult call() {
+      UserJourneyResult result =
+          new UserJourneyResult(testScriptName, Thread.currentThread().getName());
+
+      try {
+        testScript.userJourney(apiClient);
+      } catch (Exception ex) {
+        result.exceptionThrown = ex;
+      }
+      return result;
+    }
+  }
+
+  void modifyHelmValuesAndDeploy() {
+    // TODO: modify Helm template with DR Manager property overrides and do deploy
```

Comment:
Are you planning on having separate PRs for the Helm and Kube todos?

Reply:
Yes, there are separate PRs for both. I've implemented one of the Kubernetes properties (DR-1143: initial size of replica set) and added the YAML parsing code for the Helm deploy (DR-1030). There's another ticket to add more Kubernetes properties (DR-1105) and I'm hoping to finish DR-1030 once I move my environment over to the non-umbrella configuration on Monday and test it there. In general, there should be tickets for most of the TODOs in the code.